src/rules/no-confusing-set-timeout.ts

```diff
@@ -0,0 +1,81 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { createRule, getNodeName, parseJestFnCall } from './utils';
+
+interface ErrorType {
+  messageId: 'globalSetTimeout' | 'multipleSetTimeouts' | 'orderSetTimeout';
+  node: TSESTree.Node;
+}
+
+function isJestSetTimeout(node: TSESTree.Node) {
+  return getNodeName(node) === 'jest.setTimeout';
+}
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Disallow using confusing setTimeout in test',
+      recommended: false,
+    },
+    messages: {
+      globalSetTimeout: '`jest.setTimeout` should be call in `global` scope.',
+      multipleSetTimeouts:
+        'Do not call `jest.setTimeout` multiple times, as only the last call will have an effect.',
+      orderSetTimeout:
+        '`jest.setTimeout` should be placed before any other jest methods',
+    },
+    type: 'problem',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const errors: ErrorType[] = [];
+    let callJestTimeout = 0;
+    let nonJestTimeout = 0;
+
+    return {
+      CallExpression(node) {
+        const scope = context.getScope();
+        const jestFnCall = parseJestFnCall(node, context);
+
+        if (!jestFnCall) return;
```

Comment:
nit: prefer having braces

```suggestion
        if (!jestFnCall) {
          return;
        }
```

(I should setup the lint rule for this tbh 😅)

Reply:
I will solve in another PR.

Reply:
I'd prefer if you address this in this PR - if you're talking about adding the lint rule, I can do that now

Reply:
> I'd prefer if you address this in this PR - if you're talking about adding the lint rule, I've opened #1429 for that

Ok, I fixed it.